### ISSUE TITLE
Clipboard functions for x11new

### ISSUE THREAD

**Issue by Bill-Gray** [wmcbrine/PDCurses#65]
`x11new` currently lacks working clipboard functions.  If I copy `xclip.c` from

https://github.com/Bill-Gray/junk

to the `x11new` directory,  and rename it to `pdcclip.c`,  _and_ add `-lpthreads` to the libraries in the Makefile,  it compiles and the testcurs tests work (after setting `HAVE_CLIPBOARD` to 1).

I originally wrote this to add clipboard functions for the VT platform.  This code will work for both.  It also has the nice feature (in my opinion) of allowing one to use the PDC_*clipboard functions in any program,  including those not linked to the PDCurses library,  including those that _are_ linked to ncurses.

Note that by changing a couple of strings from `CLIPBOARD` to `PRIMARY`,  you can switch from setting/getting the clipboard to setting/getting the current selection.  I've not made it possible yet to make that switch programmatically,  but am strongly tempted to do so.

A question : I see you're still making improvements to the "original" `fork()`-using x11 platform.  Do you expect that to fade away now that we've got x11new to play with?  Is there still a use for the "original" platform?

**Comment by wmcbrine** [comment]
> A question : I see you're still making improvements to the "original" fork()-using x11 platform. Do you expect that to fade away now that we've got x11new to play with? Is there still a use for the "original" platform?

x11new isn't quite ready to take over yet. For one (more) thing, the problem you described in 9496141 is still present. (I found a simple fix that made the composed input work -- except that it then came in doubled.) But even once I'm satisfied with it, I'll want at least one broad release with both ports, to get x11new tested more widely.

**Comment by wmcbrine** [comment]
So... it's pretty much ready now. I fixed the composed input ( https://github.com/wmcbrine/PDCurses/commit/985e8439c1d10b1f251feaca35982a3bd10fe810 ), I redid the clipboard functions for x11new (without threads), and I wrote a "universal" (all platforms) cut-and-paste mechanism, on the curses side, to handle mouse select and paste, as well as keyboard (shift-ctrl-c and shift-ctrl-v). There are still some improvements to make, but it's at the level where the old x11 was before (plus keyboard paste, plus sdl2 has cut-and-paste support for the first time).
